WapTenantPublicAPI.psm1

```diff
@@ -1565,6 +1565,103 @@ function Grant-WAPVMNetworkAccess {
     }
 }
 
+function Set-WAPVMNetwork {
+
+    <#
+
+    .SYNOPSIS
+
+        Sets subscription available VM Networks from Azure Pack TenantPublic or Tenant API.
+
+    .PARAMETER VMNetwork
+
+        When VMNetwork is specified, only that VM Network is returned.
+
+    .PARAMETER Name
+
+        When Name is specified, the VM Network is set with said Name.
+
+    .PARAMETER Description
+
+        When Description is specified, the VM Network is set with said Description.
+
+
+
+    .EXAMPLE
+
+        PS C:\>$URL = 'https://publictenantapi.mydomain.com'
+
+        PS C:\>$creds = Get-Credential
+
+        PS C:\>Get-WAPToken -Credential $creds -URL 'https://sts.adfs.com' -ADFS
+
+        PS C:\>Connect-WAPAPI -URL $URL
+
+        PS C:\>Get-WAPSubscription -Name 'MySubscription' | Select-WAPSubscription
+
+        PS C:\>$WAPNet | Set-WAPVMNetwork -Name "MyNewName"
+
+
+
+        This will fetch all VM Networks available to the subscription.
+
+    #>
+
+    [CmdletBinding()]
+    param (
+        [parameter()]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.PSTypeName('WAP.VMNetwork')] $VMNetwork,
+
+        [parameter()]
+        [ValidateNotNullOrEmpty()]
+        [String] $Name,
+
+        [parameter()]
+        [ValidateNotNullOrEmpty()]
+        [String] $Description
+    )
+
+    process {
+        try {
+            if ($IgnoreSSL) {
+                Write-Warning -Message 'IgnoreSSL defined by Connect-WAPAPI, Certificate errors will be ignored!'
+                #Change Certificate Policy to ignore
+                IgnoreSSL
+            }
+
+            PreFlight -IncludeConnection -IncludeSubscription
+
+            $URI = '{0}:{1}/{2}/services/systemcenter/vmm/VMNetworks(ID=guid''{3}'',StampId=guid''{4}'')' -f $PublicTenantAPIUrl, $Port, $Subscription.SubscriptionId, $VMNetwork.id, $VMNetwork.stampid
+
+            Write-Verbose -Message "Constructed VM Networks URI: $URI"
+
+            if ($PSBoundParameters.ContainsKey("Name")) {
+                $VMNetwork.Name = $Name 
+            }
+
+            if ($PSBoundParameters.ContainsKey("Description")) {
+                $VMNetwork.Description = $Description 
+            }
+            
```

Comment:
if both Name and Description parameters don't have values assigned, there is no need to send in the Put request.
In that case I rather have it bail out with a warning or something

Reply:
Addressed in lines starting 1634